pkg/daemon/server/service/rater/pod_tracker.go

```diff
@@ -98,17 +99,29 @@ func (pt *PodTracker) trackActivePods(ctx context.Context) {
 }
 
 func (pt *PodTracker) updateActivePods() {
+	var wg sync.WaitGroup
+	var mu sync.Mutex
+
 	for _, v := range pt.pipeline.Spec.Vertices {
-		for i := 0; i < int(v.Scale.GetMaxReplicas()); i++ {
-			podName := fmt.Sprintf("%s-%s-%d", pt.pipeline.Name, v.Name, i)
-			podKey := pt.getPodKey(i, v.Name)
-			if pt.isActive(v.Name, podName) {
-				pt.activePods.PushBack(podKey)
-			} else {
-				pt.activePods.Remove(podKey)
-			}
+		for i := range int(v.Scale.GetMaxReplicas()) {
+			wg.Add(1)
+			go func(vertexName string, index int) {
+				defer wg.Done()
+				podName := fmt.Sprintf("%s-%s-%d", pt.pipeline.Name, vertexName, index)
+				podKey := pt.getPodKey(index, vertexName)
+				if pt.isActive(vertexName, podName) {
+					mu.Lock()
```

Comment:
Do we need to add the lock? The activePods as a `UniqueStringList`, already has RWMutex in-place.